sdk/python/kfp/v2/components/experimental/component_spec.py

```diff
@@ -14,40 +14,36 @@
 """Definitions for component spec."""
 
 from dataclasses import dataclass
-from typing import Any, Mapping, Optional, Sequence, Union
+from typing import Any, Dict, Mapping, Optional, Sequence, Union
+from pydantic import BaseModel
+import yaml
+import json
 
 from kfp.components import _structures as v1_components
 from kfp.components import _data_passing
 
 
-@dataclass
-class InputSpec:
+class InputSpec(BaseModel):
   """Component input definitions.
 
   Attributes:
-    name: The name of the input.
     type: The type of the input.
     default: Optional; the default value for the input.
   """
-  name: str
-  type: Any
+  type: Union[str, int]
```

Comment:
`Union[str, int, float, bool, dict, list]` (consider define an alias for it), and add a TODO that it will include artifact classes as well.